src/command/new/mod.rs

```diff
@@ -0,0 +1,348 @@
+}
+
+impl fmt::Display for GithubTemplate {
+  // This trait requires `fmt` with this exact signature.
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    // Write strictly the first element into the supplied output
+    // stream: `f`. Returns `fmt::Result` which indicates whether the
+    // operation succeeded or failed. Note that `write!` uses syntax which
+    // is very similar to `println!`.
+    write!(f, "{}", self.display)
+  }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Parser)]
+pub enum ProjectType {
+  ALL,
+  SUBGRAPH,
+  CLIENT,
+}
+
+impl PartialEq<String> for ProjectType {
+  fn eq(&self, other: &String) -> bool {
+    match &self {
+      ProjectType::ALL => other.to_lowercase() == "all",
+      ProjectType::SUBGRAPH => other.to_lowercase() == "subgraph",
+      ProjectType::CLIENT => other.to_lowercase() == "client",
+    }
+  }
+}
+impl PartialEq for ProjectType {
+  fn eq(&self, other: &ProjectType) -> bool {
+    match &self {
+      ProjectType::ALL => other.eq(&String::from("all")),
+      ProjectType::SUBGRAPH => other.eq(&String::from("subgraph")),
+      ProjectType::CLIENT => other.eq(&String::from("client")),
+    }
+  }
+}
+impl fmt::Display for ProjectType {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    match self {
+      ProjectType::ALL => write!(f, "all"),
+      ProjectType::SUBGRAPH => write!(f, "subgraph"),
+      ProjectType::CLIENT => write!(f, "client"),
+    }
+  }
+}
+
+impl FromStr for ProjectType {
+  type Err = saucer::Error;
+
+  fn from_str(input: &str) -> std::result::Result<Self, Self::Err> {
+    match input {
+      "all" => Ok(ProjectType::ALL),
+      "subgraph" => Ok(ProjectType::SUBGRAPH),
+      "client" => Ok(ProjectType::CLIENT),
+      _ => Err(anyhow!("Invalid Project Type")),
+    }
+  }
+}
+
+impl New {
+  pub fn run(&self, client_config: StudioClientConfig) -> Result<RoverOutput> {
+    // read_dir will always have the root element, or a count of 1
+    if read_dir(&self.path).unwrap().count() > 1 {
+      return Err(RoverError::new(anyhow!(
+        "You can only create projects in a blank folder. This is to prevent from accidentally overwriting any work."
+      )));
+    }
+
+    let client = client_config.get_reqwest_client()?;
+    if self.options.template.is_some() {
+      // User provided template id
+      self.extract_github_tarball(
+        &self.options.template.clone().unwrap().as_str(),
+        &self.path,
+        &client,
+      )?;
+    } else {
+      let template_id: String;
+      let templates = self.get_templates();
+      let should_prompt_project_type = self.options.project_type.is_none();
+      let should_prompt_language = self.options.language.is_none();
+
+      let project_type: ProjectType;
+      if should_prompt_project_type {
+        project_type = self.prompt_project_type()?;
+      } else {
+        project_type = ProjectType::from_str(&self.options.project_type.clone().unwrap())?;
+      }
```

Comment:
something like this would be a bit cleaner

```suggestion
      let project_type = self.options.project_type.unwrap_or_else(|| self.prompt_project_type()?);
```